code/client/src/pages/Show/Swap.jsx

```diff
@@ -0,0 +1,268 @@
+  const [swapAmount, setSwapAmount] = useState()
+  const [selectedTokenBalance, setSelectedTokenBalance] = useState('0')
+
+  // Loads supported tokens that are available for swap.
+  useEffect(() => {
+    const loadCryptos = async () => {
+      // TODO: this is not tokens supported by Harmony network, fetch supported tokens from Harmony network.
+      const loadedCryptos = await mockCryptos()
+      setSupportedTokens(loadedCryptos)
+    }
+
+    loadCryptos()
+  }, [setSupportedTokens])
+
+  useEffect(() => {
+    const loadTrackedTokensMetadata = async () => {
+      const trackedTokensWithMetadata = await Promise.all(currentTrackedTokens.map(async (trackedToken) => {
+        try {
+          if (trackedToken.symbol === HarmonyONE.symbol) {
+            return trackedToken
+          }
+
+          const { name, symbol, decimals } = await api.blockchain.getTokenMetadata(trackedToken)
+
+          return {
+            ...trackedToken,
+            name,
+            symbol,
+            decimals
+          }
+        } catch (ex) {
+          console.error(ex)
+        }
+      }))
+      setCurrentTrackedTokens(trackedTokensWithMetadata)
+    }
+
+    loadTrackedTokensMetadata()
+  }, [currentTrackedTokens, setCurrentTrackedTokens])
+
+  useEffect(() => {
+    const tokenBalance = getSelectedTokenComputedBalance(selectedTokenSwapFrom, tokenBalances, balance)
+
+    if (!tokenBalance) {
+      return '0'
+    }
+
+    setSelectedTokenBalance(tokenBalance.formatted)
+  }, [selectedTokenSwapFrom, tokenBalances, balance])
+
+  const swapOptions = (tokens, setSelectedToken) => tokens.map((token, index) => (
+    <Select.Option key={index} value={`${token.symbol} ${token.name}`} style={selectOptionStyle}>
+      <Button
+        type='text'
+        block
+        style={optionButtonStyle}
+        onClick={() => setSelectedToken(token)}
+      >
+        <TokenLabel token={token} />
+      </Button>
+    </Select.Option>
+  ))
+
+  const handleSearchSupportedTokens = async (value) => {
+    setSelectedTokenSwapTo({ value })
+  }
+
+  const handleSearchCurrentTrackedTokens = async (value) => {
+    setSelectedTokenSwapFrom({ value })
+  }
+
+  const onSelectTokenSwapFrom = (token) => {
+    setSelectedTokenSwapFrom({ ...token, value: token.symbol, label: <TokenLabel token={token} /> })
+  }
+
+  const onSelectTokenSwapTo = (token) => {
+    setSelectedTokenSwapTo({ ...token, value: token.symbol, label: <TokenLabel token={token} /> })
+  }
+
+  const onSwapAmountChange = (value) => {
+    const amount = trim(value)
+    if (isNaN(parseInt(amount, 10))) {
+      setSwapAmount(undefined)
+    } else {
+      setSwapAmount(amount)
+    }
+  }
+
+  const confirmSwap = useCallback(() => {
+    const swapAmountBalance = util.toBalance(swapAmount)
+    const computedTokenBalance = getSelectedTokenComputedBalance(selectedTokenSwapFrom, tokenBalances, balance)
+    const tokenBalance = util.toBalance(computedTokenBalance.formatted)
```

Comment:
balance equals tokenBalance.balance. There is no need converting back and forth only to produce the same result

Reply:
`computedTokenBalance` returns formatted balance, but we want the BN format for comparison?

Reply:
util.computeBalance returns the balance you passed in the parameters as the `balance` in its output. So if you only use `balance` field from `util.computeBalance`, the function is not doing anything. It is not guaranteed to be a BN. Also, `balance` everywhere is always referring to the raw, non-divisible amount. For ONE or ETH, it would be in wei.